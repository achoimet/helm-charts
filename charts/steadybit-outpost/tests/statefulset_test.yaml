templates:
  - statefulset.yaml
  - secrets.yaml
tests:
  - it: manifest should match snapshot
    set:
      global:
        clusterName: test
      outpost:
        key: abcdefg
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }

  - it: should apply extra pod labels
    set:
      global:
        clusterName: test
      outpost:
        key: abcdefg
      podLabels:
        some-label: some-value
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }

  - it: using image pull secrets
    set:
      global:
        clusterName: test
      image:
        pullSecrets:
          - test-pull-secret
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }
  - it: using extensions with mtls from containerpath
    set:
      global:
        clusterName: test
      outpost:
        key: abcdefg
        extraCertificates:
          path: /etc/ssl/extra-certs
        extensions:
          tls:
            hostnameVerification: strict
            clientCertificate:
              path: /etc/ssl/certs/client.crt
              key:
                path: /etc/ssl/certs/client.key
                password:
                  value: some-extension-client-key-password
            serverCertificate:
              path: /etc/ssl/certs/server.crt
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }

  - it: using extensions with mtls from secrets
    set:
      global:
        clusterName: test
      outpost:
        key: abcdefg
        extraCertificates:
          fromVolume: extra-certs
        extraVolumes:
          - name: extra-certs
            configMap:
              name: self-signed-ca
        extensions:
          tls:
            clientCertificate:
              fromSecret: some-client-secret
              key:
                password:
                  valueFrom:
                    secretKeyRef:
                      name: some-secret
                      key: some-secret-key
            serverCertificate:
              fromSecret: server.crt
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }

  - it: should add aws account id from values
    set:
      outpost:
        key: abcdefg
        aws:
          accountId: "123456789012"
    chart:
      version: 0.0.0
    asserts:
      - matchSnapshot: { }