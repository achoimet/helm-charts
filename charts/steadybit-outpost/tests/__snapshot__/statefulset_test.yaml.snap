manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-outpost
      serviceName: steadybit-outpost
      template:
        metadata:
          annotations:
            prometheus.io/path: /prometheus
            prometheus.io/port: "42899"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-outpost
            app.kubernetes.io/version: 1.1.5
            helm.sh/chart: steadybit-outpost-1.1.5
        spec:
          containers:
            - env:
                - name: STEADYBIT_AGENT_REGISTER_URL
                  value: https://platform.steadybit.io
                - name: STEADYBIT_AGENT_KEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-outpost
                - name: POD_IP
                  valueFrom:
                    fieldRef:
                      fieldPath: status.podIP
                - name: STEADYBIT_KUBERNETES_CLUSTER_NAME
                  value: test
                - name: STEADYBIT_AGENT_POD_NAME
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.name
                - name: STEADYBIT_AGENT_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: STEADYBIT_AGENT_IDENTIFIER
                  value: test/NAMESPACE/%HOSTNAME%
                - name: STEADYBIT_HTTP_ENDPOINT_PORT
                  value: "42899"
                - name: STEADYBIT_AGENT_WORKING_DIR
                  value: /tmp/steadybit-outpost
              image: steadybit/outpost:latest
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: 42899
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-outpost
              resources:
                limits:
                  cpu: 1000m
                  memory: 500Mi
                requests:
                  cpu: 125m
                  memory: 250Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /var/lib/steadybit-outpost
                  name: steadybit-outpost-state
                - mountPath: /tmp
                  name: tmp-dir
          serviceAccountName: steadybit-outpost
          volumes:
            - emptyDir:
                medium: Memory
              name: tmp-dir
      updateStrategy:
        type: RollingUpdate
      volumeClaimTemplates:
        - metadata:
            name: steadybit-outpost-state
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 200Mi
  2: |
    apiVersion: v1
    data:
      key: YWJjZGVmZw==
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    type: Opaque
using extensions with mtls from containerpath:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-outpost
      serviceName: steadybit-outpost
      template:
        metadata:
          annotations:
            prometheus.io/path: /prometheus
            prometheus.io/port: "42899"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-outpost
            app.kubernetes.io/version: 1.1.5
            helm.sh/chart: steadybit-outpost-1.1.5
        spec:
          containers:
            - env:
                - name: STEADYBIT_AGENT_REGISTER_URL
                  value: https://platform.steadybit.io
                - name: STEADYBIT_AGENT_KEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-outpost
                - name: POD_IP
                  valueFrom:
                    fieldRef:
                      fieldPath: status.podIP
                - name: STEADYBIT_KUBERNETES_CLUSTER_NAME
                  value: test
                - name: STEADYBIT_AGENT_POD_NAME
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.name
                - name: STEADYBIT_AGENT_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: STEADYBIT_AGENT_IDENTIFIER
                  value: test/NAMESPACE/%HOSTNAME%
                - name: STEADYBIT_HTTP_ENDPOINT_PORT
                  value: "42899"
                - name: STEADYBIT_AGENT_WORKING_DIR
                  value: /tmp/steadybit-outpost
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_CHAIN_FILE
                  value: /etc/ssl/certs/client.crt
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_KEY_FILE
                  value: /etc/ssl/certs/client.key
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: extensionsClientKeyPassword
                      name: RELEASE-NAME-steadybit-outpost
                - name: STEADYBIT_AGENT_EXTENSIONS_SERVER_CERT
                  value: /etc/ssl/certs/server.crt
                - name: STEADYBIT_AGENT_EXTENSIONS_HOSTNAME_VERIFICATION
                  value: strict
                - name: STEADYBIT_AGENT_EXTRA_CERTS_PATH
                  value: /etc/ssl/extra-certs
              image: steadybit/outpost:latest
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: 42899
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-outpost
              resources:
                limits:
                  cpu: 1000m
                  memory: 500Mi
                requests:
                  cpu: 125m
                  memory: 250Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /var/lib/steadybit-outpost
                  name: steadybit-outpost-state
                - mountPath: /tmp
                  name: tmp-dir
          serviceAccountName: steadybit-outpost
          volumes:
            - emptyDir:
                medium: Memory
              name: tmp-dir
      updateStrategy:
        type: RollingUpdate
      volumeClaimTemplates:
        - metadata:
            name: steadybit-outpost-state
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 200Mi
  2: |
    apiVersion: v1
    data:
      extensionsClientKeyPassword: c29tZS1leHRlbnNpb24tY2xpZW50LWtleS1wYXNzd29yZA==
      key: YWJjZGVmZw==
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    type: Opaque
using extensions with mtls from secrets:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-outpost
      serviceName: steadybit-outpost
      template:
        metadata:
          annotations:
            prometheus.io/path: /prometheus
            prometheus.io/port: "42899"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-outpost
            app.kubernetes.io/version: 1.1.5
            helm.sh/chart: steadybit-outpost-1.1.5
        spec:
          containers:
            - env:
                - name: STEADYBIT_AGENT_REGISTER_URL
                  value: https://platform.steadybit.io
                - name: STEADYBIT_AGENT_KEY
                  valueFrom:
                    secretKeyRef:
                      key: key
                      name: RELEASE-NAME-steadybit-outpost
                - name: POD_IP
                  valueFrom:
                    fieldRef:
                      fieldPath: status.podIP
                - name: STEADYBIT_KUBERNETES_CLUSTER_NAME
                  value: test
                - name: STEADYBIT_AGENT_POD_NAME
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.name
                - name: STEADYBIT_AGENT_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: STEADYBIT_AGENT_IDENTIFIER
                  value: test/NAMESPACE/%HOSTNAME%
                - name: STEADYBIT_HTTP_ENDPOINT_PORT
                  value: "42899"
                - name: STEADYBIT_AGENT_WORKING_DIR
                  value: /tmp/steadybit-outpost
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_CHAIN_FILE
                  value: /opt/steadybit/agent/etc/extensions/client/tls.crt
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_KEY_FILE
                  value: /opt/steadybit/agent/etc/extensions/client/tls.key
                - name: STEADYBIT_AGENT_EXTENSIONS_CLIENT_CERT_PASSWORD
                  valueFrom:
                    secretKeyRef:
                      key: some-secret-key
                      name: some-secret
                - name: STEADYBIT_AGENT_EXTENSIONS_SERVER_CERT
                  value: /opt/steadybit/agent/etc/extensions/server/tls.crt
              image: steadybit/outpost:latest
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: 42899
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-outpost
              resources:
                limits:
                  cpu: 1000m
                  memory: 500Mi
                requests:
                  cpu: 125m
                  memory: 250Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /var/lib/steadybit-outpost
                  name: steadybit-outpost-state
                - mountPath: /tmp
                  name: tmp-dir
                - mountPath: /opt/steadybit/agent/etc/extensions/client
                  name: extensions-tls-client
                  readOnly: true
                - mountPath: /opt/steadybit/agent/etc/extensions/server
                  name: extensions-tls-server
                  readOnly: true
                - mountPath: /opt/steadybit/outpost/etc/extra-certs
                  name: extra-certs
          serviceAccountName: steadybit-outpost
          volumes:
            - emptyDir:
                medium: Memory
              name: tmp-dir
            - name: extensions-tls-client
              secret:
                secretName: some-client-secret
            - name: extensions-tls-server
              secret:
                secretName: server.crt
            - configMap:
                name: self-signed-ca
              name: extra-certs
      updateStrategy:
        type: RollingUpdate
      volumeClaimTemplates:
        - metadata:
            name: steadybit-outpost-state
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 200Mi
  2: |
    apiVersion: v1
    data:
      key: YWJjZGVmZw==
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    type: Opaque
using image pull secrets:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-outpost
      serviceName: steadybit-outpost
      template:
        metadata:
          annotations:
            prometheus.io/path: /prometheus
            prometheus.io/port: "42899"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: steadybit-outpost
            app.kubernetes.io/version: 1.1.5
            helm.sh/chart: steadybit-outpost-1.1.5
        spec:
          containers:
            - env:
                - name: STEADYBIT_AGENT_REGISTER_URL
                  value: https://platform.steadybit.io
                - name: POD_IP
                  valueFrom:
                    fieldRef:
                      fieldPath: status.podIP
                - name: STEADYBIT_KUBERNETES_CLUSTER_NAME
                  value: test
                - name: STEADYBIT_AGENT_POD_NAME
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.name
                - name: STEADYBIT_AGENT_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: STEADYBIT_AGENT_IDENTIFIER
                  value: test/NAMESPACE/%HOSTNAME%
                - name: STEADYBIT_HTTP_ENDPOINT_PORT
                  value: "42899"
                - name: STEADYBIT_AGENT_WORKING_DIR
                  value: /tmp/steadybit-outpost
              image: steadybit/outpost:latest
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health
                  port: 42899
                initialDelaySeconds: 30
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: steadybit-outpost
              resources:
                limits:
                  cpu: 1000m
                  memory: 500Mi
                requests:
                  cpu: 125m
                  memory: 250Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts:
                - mountPath: /var/lib/steadybit-outpost
                  name: steadybit-outpost-state
                - mountPath: /tmp
                  name: tmp-dir
          imagePullSecrets:
            - name: test-pull-secret
          serviceAccountName: steadybit-outpost
          volumes:
            - emptyDir:
                medium: Memory
              name: tmp-dir
      updateStrategy:
        type: RollingUpdate
      volumeClaimTemplates:
        - metadata:
            name: steadybit-outpost-state
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 200Mi
  2: |
    apiVersion: v1
    data: null
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-outpost
        app.kubernetes.io/version: 1.1.5
        helm.sh/chart: steadybit-outpost-1.1.5
      name: RELEASE-NAME-steadybit-outpost
      namespace: NAMESPACE
    type: Opaque
